src/solc_0.7/MultiGiveaway/ClaimERC721AndERC1155WithERC20.sol

```diff
@@ -0,0 +1,122 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.7.5;
+
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../common/Interfaces/IERC721Extended.sol";
+
+contract ClaimERC721AndERC1155WithERC20 {
+    bytes32 internal _merkleRoot;
+    IERC1155 internal immutable _asset;
+    IERC20 internal immutable _erc20Token;
+    IERC721Extended internal immutable _land;
+    address internal immutable _assetsHolder;
+    address internal immutable _landHolder;
+    address internal immutable _erc20TokenHolder;
+    event ClaimedAssetsAndLandsWithERC20(
+        address to,
+        uint256[] assetIds,
+        uint256[] assetValues,
+        uint256[] landIds,
+        uint256 erc20TokenAmount
+    );
+
+    constructor(
+        IERC1155 asset,
+        IERC721Extended land,
+        IERC20 erc20Token,
+        address assetsHolder,
+        address landHolder,
+        address erc20TokenHolder
+    ) {
+        _asset = asset;
+        _land = land;
+        _erc20Token = erc20Token;
+        if (assetsHolder == address(0)) {
+            assetsHolder = address(this);
+        }
+        if (landHolder == address(0)) {
+            landHolder = address(this);
+        }
+        if (erc20TokenHolder == address(0)) {
+            erc20TokenHolder = address(this);
+        }
```

Comment:
The same comment applies to ERC721 ERC1155 claim contract, but I am not sure for  use case where the holder will be different among tokens.
It does not increase much the complexity so might be fine to have that feature though